openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -223,7 +223,9 @@ def is_flag_active(self, flag_name, check_before_waffle_callback=None, flag_unde
                 if value is None:
                     value = flag_is_active(get_request(), namespaced_flag_name)
 
-            self._cached_flags[namespaced_flag_name] = value
+            # Don't cache the result if it might have been affected by the callback
```

Comment:
@robrap this is the line of code that is necessary to allow course Waffle flags to work with different values. Let's discuss tomorrow.